src/DI/ServiceDefinition.php

```diff
@@ -210,19 +210,19 @@ public function getTag($tag)
 
 
 	/**
-	 * @param  bool
+	 * @param  bool|string|string[]
 	 * @return self
 	 */
 	public function setAutowired($state = TRUE)
 	{
 		call_user_func($this->notifier);
-		$this->autowired = (bool) $state;
+		$this->autowired = is_string($state) || is_array($state) ? (array) $state : (bool) $state;
 		return $this;
 	}
 
 
 	/**
-	 * @return bool
+	 * @return bool|string[]
 	 */
 	public function isAutowired()
```

Comment:
Method with `isX` name, that returns `bool` and `string` looks really confusing to me.

I'd keep preferred autowired types and TRUE/FALSE state separated.